src/runtime/graph/graph_runtime.cc

```diff
@@ -232,6 +241,13 @@ class GraphRuntime : public ModuleNode {
     TVMOpParam param;
     // inputs
     std::vector<NodeEntry> inputs;
+    // device_type is used to indicate where the node should be scheduled to.
+    // TODO(zhiics) device_type is defaulted to CPU for transition purpose only
+    // because it will have random value otherwise. Using this default value is
+    // to make sure homogeneous execution works correctly. It will be removed
+    // when we add the compiler pass as we will read the serialized value from
+    // json for all execution.
+    DLDeviceType device_type{kDLCPU};
```

Comment:
Let us not put device_type here, instead, introduce a device_index column attribute, just like storage_id to indicate device assignment

Reply:
If the column do not exist, fallback to default(0 primary context passed in)

Reply:
@tqchen So we use ctxs_[0] as the default, right?

Reply:
I think that is a reasonable choice